v2/RNHealthKitCore/ReactNativeBridge/RNHealthKitWrapper.swift

```diff
@@ -0,0 +1,121 @@
+import Foundation
+import HealthKit
+
+@objc(RNHealthKitWrapper)
+class RNHealthKitWrapper: NSObject {
+    var core: HealthKitCore?
+
+    @objc
+    func initHealthKit(_ read: Array<String>, write: Array<String>, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
```

Comment:
potential-issue: looks like is not possible to request `workout` permissions since this won't try to create it. I think you might have to go for the imperative code approach here
```swift
var readTypes: [any HealthKitType] = []
for identifier in read {
 guard let readType = QuantityType(rawValue: identifier) ?? WorkoutType(rawValue: identifier) else { continue }
 readTypes.append(readType)
}
```

Reply:
Workout is not properly exported yet, currently working on that 😅